wal/wal.go

```diff
@@ -265,10 +266,12 @@ func (w *WAL) ReadAll() (metadata []byte, state raftpb.HardState, ents []raftpb.
 			return nil, state, nil, fmt.Errorf("unexpected block type %d", rec.Type)
 		}
 	}
-	if err != io.EOF {
+	// we must read all of the entries if WAL is opened in write mode.
+	if err != io.EOF && w.f != nil {
```

Comment:
why do we need to care about other errors? it is possible that we cannot read all data or decode all data. it is fine.